packages/cli/src/cmds/beacon/initBeaconState.ts

```
@@ -187,6 +188,15 @@ export async function initBeaconState(
     // Convert to `Uint8Array` to avoid unexpected behavior such as `Buffer.prototype.slice` not copying memory
     stateBytes = new Uint8Array(stateBytes.buffer, stateBytes.byteOffset, stateBytes.byteLength);
     const anchorState = getStateTypeFromBytes(chainForkConfig, stateBytes).deserializeToViewDU(stateBytes);
+    // Validate genesis state root
+    const stateRoot = toRootHex(anchorState.hashTreeRoot());
+    const expectedRoot = getGenesisStateRoot(args.network || defaultNetwork);
+    if (expectedRoot && stateRoot !== expectedRoot) {
+      logger.error("Genesis state root mismatch", {
+        expected: expectedRoot,
+        found: stateRoot,
+      });
```

we should throw an error here instead